framework/include/base/MooseApp.h

```diff
@@ -1096,7 +1101,7 @@ class MooseApp : public ConsoleStreamInterface,
   const std::shared_ptr<Parallel::Communicator> _comm;
 
   /// Input file names used
-  std::vector<std::string> _input_filenames;
+  std::vector<FileName> _input_filenames;
```

Comment:
The type checking should stay in a separate PR.  This PR isn't needed with WASP #include.  Sorry....  I didn't realize that when I requested this.